File: src/main/java/com/fifo/ticketing/domain/user/service/AdminService.java
package com.fifo.ticketing.domain.user.service;

import com.fifo.ticketing.domain.user.dto.UserDto;
import com.fifo.ticketing.domain.user.entity.Role;
import com.fifo.ticketing.domain.user.entity.User;
import com.fifo.ticketing.domain.user.mapper.UserMapper;
import com.fifo.ticketing.domain.user.repository.UserRepository;
import com.fifo.ticketing.global.exception.ErrorCode;
import com.fifo.ticketing.global.exception.ErrorException;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class AdminService {

    private final UserRepository userRepository;

    @Transactional(readOnly = true)
    public Page<UserDto> getAllUsers(Pageable pageable) {
        Page<User> allUserInfo = userRepository.findAll(pageable);
        return UserMapper.toUserDtoPage(allUserInfo);
    }

    @Transactional(readOnly = true)
    public Page<UserDto> getUsersByName(Pageable pageable, String name) {
        Page<User> byUsernameContaining = userRepository.findByUsernameContaining(name, pageable);
        return UserMapper.toUserDtoPage(byUsernameContaining);
    }

    @Transactional(readOnly = true)
    public Page<UserDto> getUsersByRole(Pageable pageable, Role role) {
        Page<User> byRole = userRepository.findByRole(role, pageable);
        return UserMapper.toUserDtoPage(byRole);
    }

    @Transactional(readOnly = true)
    public Page<UserDto> getUsersByRoleAndName(Pageable pageable, Role role, String name) {
        Page<User> byUsernameContainingAndRole = userRepository.findByUsernameContainingAndRole(
            name, role, pageable);
        return UserMapper.toUserDtoPage(byUsernameContainingAndRole);
    }

    @Transactional
    public void updateUserStatus(Long userId) {
        User user = userRepository.findById(userId).orElseThrow(() ->
            new ErrorException(ErrorCode.NOT_FOUND_MEMBER)
        );
        user.updateBlockedState();
    }

}

File: src/main/java/com/fifo/ticketing/domain/user/service/AdminService.kt
package com.fifo.ticketing.domain.user.service

import com.fifo.ticketing.domain.user.dto.UserDto
import com.fifo.ticketing.domain.user.entity.Role
import com.fifo.ticketing.domain.user.mapper.UserMapper.toUserDtoPage
import com.fifo.ticketing.domain.user.repository.UserRepository
import com.fifo.ticketing.global.exception.ErrorCode
import com.fifo.ticketing.global.exception.ErrorException
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class AdminService(
    private val userRepository: UserRepository
) {

    @Transactional(readOnly = true)
    fun getAllUsers(pageable: Pageable): Page<UserDto> {
        val allUserInfo = userRepository.findAll(pageable)
        return toUserDtoPage(allUserInfo)
    }

    @Transactional(readOnly = true)
    fun getUsersByName(pageable: Pageable, name: String): Page<UserDto> {
        val byUsernameContaining = userRepository.findByUsernameContaining(name, pageable)
        return toUserDtoPage(byUsernameContaining)
    }

    @Transactional(readOnly = true)
    fun getUsersByRole(pageable: Pageable, role: Role): Page<UserDto> {
        val byRole = userRepository.findByRole(role, pageable)
        return toUserDtoPage(byRole)
    }

    @Transactional(readOnly = true)
    fun getUsersByRoleAndName(pageable: Pageable, role: Role, name: String): Page<UserDto> {
        val byUsernameContainingAndRole = userRepository.findByUsernameContainingAndRole(
            name, role, pageable
        )
        return toUserDtoPage(byUsernameContainingAndRole)
    }

    @Transactional
    fun updateUserStatus(userId: Long) {
        val user = userRepository.findById(userId).orElseThrow {
            ErrorException(
                ErrorCode.NOT_FOUND_MEMBER
            )
        }
        user.updateBlockedState()
    }
}
